src/main/scala/org/fortysevendeg/sparksftp/ReadingSFTPHadoopApp.scala

```diff
@@ -22,6 +25,13 @@ object ReadingSFTPHadoopApp extends IOApp {
       defaultSparkConf: SparkConf = createSparkConfWithSFTPSupport(config)
```

Comment:
I'd organize this `for` comprehension in order to ease the reading. We're inside of an `IO` and these methods look like they're effects. Why not create some auxiliary `def`s and join them here?

```scala
def createSparkSession(config: ReadingSFTPConfig): IO[SparkSession] = IO {
  val defaultSparkConf: SparkConf = createSparkConfWithSFTPSupport(config)
    .set("fs.sftp.impl", "org.apache.hadoop.fs.sftpwithseek.SFTPFileSystem")
    .set("fs.sftp.impl.disable.cache", "true")
  getSparkSessionWithHive(defaultSparkConf)
}

def userPath(sftpConfig: SFTPConfig): String =
  s"sftp://${sftpConfig.sftpUser}:${sftpConfig.sftpPass}@${sftpConfig.sftpHost}" + s":${sftpConfig.sftpUserPath}"

def salariesPath(sftpConfig: SFTPConfig): String =
  s"sftp://${sftpConfig.sftpUser}:${sftpConfig.sftpPass}@${sftpConfig.sftpHost}" + s":${sftpConfig.sftpSalaryPath}"

def readDataFrames(session: SparSession, sftpConfig: SFTPConfig): IO[(DataFrame, DataFrame)] =
  IO((dataframeFromCSV(sparkSession, usersPath(sftpConfig)), dataframeFromCSV(sparkSession, salariesPath(sftpConfig))))
  
def persistAndReadUserData(session: SparSession, users: DataFrame, salaries: DataFrame): IO[(DataFrame, DataFrame, DataFrame)] = IO {
  HiveUserData.persistUserData(sparkSession, users, salaries)
  HiveUserData.readUserData(sparkSession)
}

def salariesFromHive(userSalaries: DataFrame): IO[Dataset[Row]]] = IO {
  val newSalaries = userSalaries.withColumn("new_salary", (userSalaries("salary") * 1.1).cast(IntegerType))
  SparkUtils.persistDataFrame(sparkSession, newSalaries, "user_new_salary")
  sparkSession.sql("select name,salary from user_new_salary")
}

def toCompressedCSV(userData: DataFrame, salariesData: DataFrame, userNewSalaries: DataFrame, sftConfig: SFTPConfig): IO[Unit] = IO {
 dataframeToCompressedCsv(userData, s"${usersPath(sftpConfig)}_output")
 dataframeToCompressedCsv(salariesData, s"${salariesPath(sftpConfig)}_output")
 dataframeToCompressedCsv(userNewSalaries, s"${salariesPath(sftpConfig)}_transformed_output")
}

for {
  config                                 <- setupConfig
  session                                <- createSparkSession(config)
  (users, salaries)                      <- readDataFrames(session, config)
  (userData, salariesData, userSalaries) <- persistAndReadUserData(session, users, salaries)
  userNewSalaries                        <- salariesFromHive(userSalaries)
  _                                      <- toCompressedCSV(userData, salariesData, userNewSalaries)
} yield ExitCode.Success
```

Reply:
Thanks a lot for the suggestions, @fedefernandez .

Changed in 5826e18f341f76b7d4ab8b6698ab302f8f7e52e8